packages/gestalt/src/Dropdown.js

```diff
@@ -1,73 +1,116 @@
 // @flow strict
-import type { Node } from 'react';
-
-import { Children, cloneElement, useState } from 'react';
+import { Children, cloneElement, type Node, useState } from 'react';
 import PropTypes from 'prop-types';
 import Box from './Box.js';
 import Popover from './Popover.js';
 import Layer from './Layer.js';
 import DropdownItem from './DropdownItem.js';
+import DropdownLink from './DropdownLink.js';
 import DropdownSection from './DropdownSection.js';
 import { DropdownContextProvider } from './DropdownContext.js';
 import { type Indexable, UnsafeIndexablePropType } from './zIndex.js';
 import { type DirectionOptionType } from './utils/keyboardNavigation.js';
-import { type AbstractEventHandler } from './AbstractEventHandler.js';
 import { ESCAPE, SPACE, TAB, ENTER, UP_ARROW, DOWN_ARROW } from './keyCodes.js';
-import { type OptionObject } from './MenuOption.js';
+
+const KEYS = {
+  UP: -1,
+  DOWN: 1,
+  ENTER: 0,
+};
+
+const dropdownItemDisplayNames = ['DropdownItem', 'DropdownLink'];
+
+function getChildrenOptions(childrenArray) {
+  return childrenArray.reduce((accumulatedChildren, currentChild) => {
+    const {
+      props: { children: currentItemChildren },
+      type: { displayName },
+    } = currentChild;
+
+    if (currentItemChildren && displayName === 'DropdownSection') {
```

Comment:
It looks like the keyboard navigation is broken when Dropdown.Section only has one item (guessing the indexing gets off maybe?)